pkg/apis/starlingx/v1/zz_generated.deepequal.go

```diff
@@ -898,6 +898,14 @@ func (in *MatchBMInfo) DeepEqual(other *MatchBMInfo) bool {
 		}
 	}
 
+	if (in.Type == nil) != (other.Type == nil) {
+		return false
+	} else if in.Type != nil {
```

Comment:
Shouldn't this need to check that other.Type is not nil otherwise it could generate an error.  I know this follows the other fields, but I'm not sure why it doesn't also have the possibility of dereferencing a nil value.

Reply:
The very first check in the function is:
        if other == nil {
                return false
        }

Reply:
It is the dereference of *other.Type that I am concerned about.  I don't see that checked anywhere.

Reply:
Isn't that caught by the check on line 901? If one is nil and one is not, then it returns false. So if in.Type is non-nil at 903, then other.Type must also be non-nil

Reply:
Yes.  Sorry, you are correct.